plugins/native_dio_adapter/lib/src/native_adapter.dart

```diff
@@ -17,14 +17,29 @@ import 'cupertino_adapter.dart';
 /// make HTTP requests.
 class NativeAdapter implements HttpClientAdapter {
   NativeAdapter({
+    CronetEngine Function()? createCronetEngine,
+    URLSessionConfiguration Function()? createCupertinoConfiguration,
+    @Deprecated(
+      'Use createCronetEngine instead. '
+      'This will cause platform exception on iOS/macOS platforms. '
+      'This will be removed in v2.0.0',
+    )
     CronetEngine? androidCronetEngine,
+    @Deprecated(
+      'Use createCupertinoConfiguration instead. '
+      'This will cause platform exception on the Android platform. '
+      'This will be removed in v2.0.0',
+    )
     URLSessionConfiguration? cupertinoConfiguration,
   }) {
     if (Platform.isAndroid) {
```

Comment:
Unrelated to this PR, but I guess we could make use of https://pub.dev/packages/platform here later to enable better testing in the future